### PR TITLE
 load balance ICDS warehouse reads

### PR DESCRIPTION
**Pull request by snopoke**
@dimagi/scale-team 

buddy @millerdev

### CODE REVIEW

File: custom/icds_reports/tasks.py [hunk 1]
```diff
@@ -48,8 +49,12 @@ def move_ucr_data_into_aggregation_tables(date=None, intervals=2):
 
     monthly_dates.append(date)
 
-    if hasattr(settings, "ICDS_UCR_DATABASE_ALIAS") and settings.ICDS_UCR_DATABASE_ALIAS:
-        with connections[settings.ICDS_UCR_DATABASE_ALIAS].cursor() as cursor:
+    try:
+        db_alias = connection_manager.get_django_db_alias(ICDS_UCR_ENGINE_ID)
+    except KeyError:
+        pass
```
Comment (czue) [inline custom/icds_reports/tasks.py]: would there be any value in extracting this out into a function? saw it in a few places

**Review by czue — APPROVED** [review]
not sure if someone else should review since I don't have too much context on this code but looks good to me.

File: custom/icds_reports/tasks.py (hunk 1 above)
Comment (snopoke) [reply custom/icds_reports/tasks.py]: 6adac74

**Comment by dannyroberts** [comment]
related?
```
======================================================================
ERROR: test suite for <module 'custom.icds_reports.tests' from '/mnt/commcare-hq-ro/custom/icds_reports/tests/__init__.pyc'>
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/vendor/lib/python2.7/site-packages/nose/suite.py", line 209, in run
    self.setUp()
  File "/vendor/lib/python2.7/site-packages/nose/suite.py", line 292, in setUp
    self.setupContext(ancestor)
  File "/vendor/lib/python2.7/site-packages/nose/suite.py", line 315, in setupContext
    try_run(context, names)
  File "/vendor/lib/python2.7/site-packages/nose/util.py", line 471, in try_run
    return func()
  File "/mnt/commcare-hq-ro/custom/icds_reports/tests/__init__.py", line 87, in setUpModule
    table = metadata.tables[table_name]
KeyError: 'config_report_icds-cas_static-child_health_cases_a46c129f'
```

**Review by calellowitz — APPROVED** [review]
looks good to me. do not have a ton of context on a lot of db routing internals but the changes seem right, and i like the idea of load balancing these better

**Review by millerdev — APPROVED** [review]

**Review by emord — APPROVED** [review]
Looks good to me. Will wait to merge until @snopoke is back online

**Pull request merged**